test/unit/green-flag.test.jsx

```diff
@@ -0,0 +1,41 @@
+/* eslint-env jest */
+const React = require('react'); // eslint-disable-line no-unused-vars
+const {shallow} = require('enzyme');
+const GreenFlag = require('../../src/containers/green-flag'); // eslint-disable-line no-unused-vars
+const renderer = require('react-test-renderer');
+const VM = require('scratch-vm');
+
+describe('GreenFlag', () => {
+    let vm;
+    beforeEach(() => {
+        // @todo: Ask scratch team their thoughts on mocking the VM.
+        vm = new VM();
+    });
+
+    test('renders active state', () => {
+        const component = renderer.create(
+            <GreenFlag active={true} vm={vm}/>
+        );
+        expect(component.toJSON()).toMatchSnapshot();
+    });
+
+    test('renders inactive state', () => {
+        const component = renderer.create(
+            <GreenFlag active={false} vm={vm}/>
+        );
+        expect(component.toJSON()).toMatchSnapshot();
+    });
+
+    test('triggers onClick when active', () => {
+        const onClick = jest.fn();
+        const componentShallowWrapper = shallow(
+            <GreenFlag active={true} onClick={onClick} vm={vm}/>
```

Comment:
Ah I think I understand why the test passes. It is a quirk with the way we are passing props in. Because you passed in a `onChange`, and the render code looks like this:
```
            <GreenFlagComponent
                active={this.state.projectRunning}
                onClick={this.handleClick}
                {...props}
            />
```
one of those props in `{...props}` is actually another onClick, which because of the way object spreads work, will override the other onClick.

@rschamp it may be a good idea to always pass in the object spread _before_ the other props, to make sure this kind of thing doesn't happen. i.e. 
```
            <GreenFlagComponent
                {...props}
                active={this.state.projectRunning}
                onClick={this.handleClick}
            />
```

To check this, you might try changing the green flag container to the above ^ and then see the test should fail.